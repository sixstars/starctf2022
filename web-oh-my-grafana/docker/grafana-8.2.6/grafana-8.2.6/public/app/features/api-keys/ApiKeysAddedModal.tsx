import React from 'react';
import { css } from '@emotion/css';
import { Alert, Field, Modal, useStyles2 } from '@grafana/ui';
import { GrafanaTheme2 } from '@grafana/data';

export interface Props {
  onDismiss: () => void;
  apiKey: string;
  rootPath: string;
}

export function ApiKeysAddedModal({ onDismiss, apiKey, rootPath }: Props): JSX.Element {
  const styles = useStyles2(getStyles);
  return (
    <Modal title="API Key Created" onDismiss={onDismiss} onClickBackdrop={onDismiss} isOpen>
      <Field label="Key">
        <span className={styles.label}>{apiKey}</span>
      </Field>

      <Alert severity="info" title="You will only be able to view this key here once!">
        It is not stored in this form, so be sure to copy it now.
      </Alert>

      <p className="text-muted">You can authenticate a request using the Authorization HTTP header, example:</p>
      <pre className={styles.small}>
        curl -H &quot;Authorization: Bearer {apiKey}&quot; {rootPath}/api/dashboards/home
      </pre>
    </Modal>
  );
}

function getStyles(theme: GrafanaTheme2) {
  return {
    label: css`
      padding: ${theme.spacing(1)};
      background-color: ${theme.colors.background.secondary};
      border-radius: ${theme.shape.borderRadius()};
    `,
    small: css`
      font-size: ${theme.typography.bodySmall.fontSize};
      font-weight: ${theme.typography.bodySmall.fontWeight};
    `,
  };
}
