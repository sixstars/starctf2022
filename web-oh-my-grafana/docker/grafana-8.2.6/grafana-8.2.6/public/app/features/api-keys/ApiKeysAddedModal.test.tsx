import React from 'react';
import { shallow } from 'enzyme';
import { ApiKeysAddedModal, Props } from './ApiKeysAddedModal';

const setup = (propOverrides?: object) => {
  const props: Props = {
    onDismiss: jest.fn(),
    apiKey: 'api key test',
    rootPath: 'test/path',
  };

  Object.assign(props, propOverrides);

  const wrapper = shallow(<ApiKeysAddedModal {...props} />);

  return {
    wrapper,
  };
};

describe('Render', () => {
  it('should render component', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
