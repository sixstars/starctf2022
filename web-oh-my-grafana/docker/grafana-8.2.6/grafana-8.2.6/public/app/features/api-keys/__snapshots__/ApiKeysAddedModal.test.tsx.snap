// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<Modal
  isOpen={true}
  onClickBackdrop={[MockFunction]}
  onDismiss={[MockFunction]}
  title="API Key Created"
>
  <Field
    label="Key"
  >
    <span
      className="css-ypoxjz"
    >
      api key test
    </span>
  </Field>
  <Alert
    severity="info"
    title="You will only be able to view this key here once!"
  >
    It is not stored in this form, so be sure to copy it now.
  </Alert>
  <p
    className="text-muted"
  >
    You can authenticate a request using the Authorization HTTP header, example:
  </p>
  <pre
    className="css-omfua5"
  >
    curl -H "Authorization: Bearer 
    api key test
    " 
    test/path
    /api/dashboards/home
  </pre>
</Modal>
`;
